### ISSUE TITLE
input-hog profile accept failed for XX:XX:XX

### ISSUE THREAD

**Issue by WhyNotHugo** [bluez/bluez#456]
I'm trying to connect to a bluetooth mouse. Bluetooth is otherwise working on this system; I can use headphones just fine.

I'm connecting using `bluetoothctl` to keep things simple (also, blueman fails to connect entirely).

My basic process is:

- `bluetoothctl`
- `scan on`
- `connect XX:XX:XX...`
- I get a prompt to confirm pairing. I type `yes` and hit enter.
- `bluetoothctl` now indicates: `Bonded: yes` and `Paired: yes`.

However, the mouse don't work (e.g.: moving it doesn't move the cursor). `dmesg` provides nothing of use, but `/var/log/messages` shows:

    daemon.err bluetoothd[19440]: src/service.c:service_accept() input-hog profile accept failed for XX:XX:XX:XX:XX:XX

I've no idea what this means tho. Have I hit a bug, or is this an issue with my setup?

**Comment by WhyNotHugo** [comment]
Huh, pairing the mouse with the logitech receiver (using [solaar](https://github.com/pwr-Solaar/Solaar)) prompts me to click left/right in a specified sequence in order to pair.

I wonder if support for something similar is required via bluetooth, or if that's just the receiver.

The device is  a "Logitech MX Master 3B".

**Comment by icedream** [comment]
I have run into this issue as well, the mouse will pair and connect just fine, but this exact error message will pop up twice during a second after connection and then no inputs will work.

In my case this is the Logitech G604.

**Comment by terrancewong** [comment]
fixed / workaround found. check #531

**Comment by WhyNotHugo** [comment]
The fix in https://github.com/bluez/bluez/issues/531#issuecomment-1913058753 works for me.

I'm not entirely sure if this is the same issue or not; seems to be different hardware, but the problem is the same module not being loaded in time.

**Comment by honkey48** [comment]
I had this problem with bluetooth mice. laptor with 3 distros and all was well. installed Nobira and it was ok. The other distros, various flavors of Debian, lost bluetooth mouse. They connected but no action. Someting from the Nobira install "bled in the other distros (separate partitions) cause bluetooth mouse fail. Can anyone venture a guess as to what happened? The above worked.

**Comment by devpaps** [comment]
I ran into the same issue with mu Logitech MX Master mouse on Fedora 43 with bluez 5.86. The mouse would connect fine to the bluetooth but i couldn't move the cursor. The cause was that the `uhid` kernel module was not being loaded at boot, causing `input-hog` to fail when the mouse tried to connect. I think i accidentally removed it...

This is my version of Bluez:
```
$ rpm -q bluez
bluez-5.86-4.fc43.x86_64
```

I had to re-create this service file:
`sudo nvim /etc/systemd/system/bluetooth-uhid.service `

And then add this to the file:
```
[Unit]
Description=Ensure uhid is loaded before bluetooth
Before=bluetooth.service
DefaultDependencies=no

[Service]
Type=oneshot
ExecStart=/sbin/modprobe uhid
RemainAfterExit=yes

[Install]
WantedBy=bluetooth.target
```
Then i ran these commands:
`sudo systemctl daemon-reload`
`sudo systemctl enable bluetooth-uhid.service`

Now i rebooted the laptop and the mouse started working again. I hope this helps. 🤞🏻

**Comment by WhyNotHugo** [comment]
I wonder if the error message for this particular scenario could mention "please check that the uhid module is properly loaded".

This seems to be a common error path, and I can't see an obvious solution better than that.

**Comment by devpaps** [comment]
Yes, i totally agree on that it should have a better error message. The `uhid` really went under my radar for a couple of days of debugging.
